runtime-sdk/modules/evm/src/backend.rs

```diff
@@ -471,8 +486,13 @@ impl<'ctx, 'backend, 'config, C: TxContext, Cfg: Config> StackState<'config>
     }
 
     fn inc_nonce(&mut self, address: H160) -> Result<(), ExitError> {
-        // Do not increment the origin nonce as that has already been handled by the SDK.
-        if address == self.origin() {
+        let ctx = self.backend.ctx.borrow_mut();
+
+        // Do not increment the origin nonce as that has already been handled by the SDK. But do
```

Comment:
Have no idea where this is handled. I only see one call to `Cfg::Accounts::inc_nonce` and that is few lines below..

Reply:
You mean the actual nonce increment performed by the SDK? The accounts module is handling nonce checks and increments in the authentication handler.

Reply:
I see. The module is using `checked_add`, that's why it is not calling `inc_nonce` (or `inc_nonces`, if implemented).

Not related to this PR, but `is_empty` calls `basic` twice.